mol_generator.py

```diff
@@ -0,0 +1,189 @@
+        a1 = b.atom1.index - 1
+        a2 = b.atom2.index - 1
+        t.append(_bond.format(a1=a1, a2=a2, order=b.order))
+    t.append('molecules[i] = molecule;')
+    t.append('++i;')
+    t = '\n'.join(t)
+    t = textwrap.indent(t, '    ')
+
+    return '\n{\n' + t + '\n}\n'
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('template_file')
+
+opts = parser.parse_args()
+
+git_hash = subprocess.check_output(
+    ['git', 'log', '-n', '1', '--pretty=format:%H', opts.template_file])
+git_hash = git_hash.decode().strip()[:20]
+template_dir, template_base_name = os.path.split(opts.template_file)
+
+total = structure.count_structures(opts.template_file)
+body = ''
+
+with structure.StructureReader(opts.template_file) as r:
+    for st in r:
+        mol_def = get_mol_def(st)
+        body += mol_def
+body = textwrap.indent(body, '    ')
+
+header = f"""
+#pragma once
+
+///
+// Find the templates for coordinate generation
+//
+// Autogenerated, do not edit.
+//
+//   $SCHRODINGER/run {os.path.basename(__file__)} {template_base_name}
+//
+// generated using {template_base_name} version {git_hash}.
+//
+
+#include <vector>
+
+class sketcherMinimizerMolecule;
+namespace schrodinger {{
+
+std::vector<sketcherMinimizerMolecule*> coordgen_templates();
+// const std::array<sketcherMinimizerMolecule const*, {total}>& coordgen_templates();
+}}
+"""
+
+implementation = f"""
+///
+// Find the templates for coordinate generation
+//
+// Autogenerated, do not edit.
+//
+//   $SCHRODINGER/run {os.path.basename(__file__)} {template_base_name}
+//
+// generated using {template_base_name} version {git_hash}.
+//
+
+#include "{NAME}.h"
+
+#include <array>
+
+#include "sketcherMinimizerMolecule.h"
+#include "sketcherMinimizerAtom.h"
+#include "sketcherMinimizerBond.h"
+
+using std::array;
+using std::tuple;
+
+template <typename T>
+void add_atoms(sketcherMinimizerMolecule* molecule, const T& atoms)
+{{
+    for (const auto& a: atoms) {{
+        auto atom = molecule->addNewAtom();
+        atom->setAtomicNumber(std::get<0>(a));
+        atom->setCoordinates(sketcherMinimizerPointF(std::get<1>(a), std::get<2>(a)));
+    }}
+}}
+
+template <typename T>
+void add_bonds(sketcherMinimizerMolecule* molecule, const T& bonds)
+{{
+    for (const auto& b: bonds) {{
+        auto* from_atom = molecule->getAtoms().at(b[0]);
+        auto* to_atom = molecule->getAtoms().at(b[1]);
+        auto bond = molecule->addNewBond(from_atom, to_atom);
+        bond->setBondOrder(b[2]);
+    }}
+}}
+
+std::vector<sketcherMinimizerMolecule*> create_templates()
+{{
+    std::vector<sketcherMinimizerMolecule*> molecules({total});
+    size_t i = 0;
+
+    {body}
+
+    return molecules;
+}}
+
+namespace schrodinger {{
+std::vector<sketcherMinimizerMolecule*> coordgen_templates()
+{{
+    static const auto t = create_templates();
+    return t;
+}}
+}}
+
+"""
+
+header_path = os.path.join(template_dir, f'{NAME}.h')
+with open(header_path, 'w') as fh:
+    fh.write(header)
```

Comment:
I think you need to format `header`with the number of templates.

Reply:
It's an fstring, so it gets those from the scope.